pybop/parameters/parameter.py

```diff
@@ -248,6 +250,20 @@ def remove(self, parameter_name):
         # Remove the parameter
         self.param.pop(parameter_name)
 
+    def join(self, parameters=None):
+        """
+        Join two Parameters objects into the first by copying across each Parameter.
+
+        Parameters
+        ----------
+        parameters : pybop.Parameters
+        """
+        for param in parameters:
+            if param not in self.param.values():
+                self.add(param)
+            else:
+                print(f"Discarding duplicate {param.name}.")
```

Comment:
Context: I am coming from the MultiFitting PR. If I understand this correctly, if a user provides the same parameter for 2 problems in the MultiFitting, only the first instance will make it through while the others will be discarded. This is fine if the users gives us the exact same 2 parameters (with bounds and initial values) but otherwise I think it would cause issues. I can think of 2 ways around this:

1. Throw an error if a parameter appears twice with different bounds/initial values.
2. Combine the two parameters somehow (take intersection between domains of validity?) and throw a warning to let the user know.

Reply:
Thanks for the feedback @brosaplanella! At the moment, it is option 1 because adding a parameter with the same name will return an error (see [parameter.py line 223](https://github.com/pybop-team/PyBOP/blob/358-passing-inputs/pybop/parameters/parameter.py#L223)) - even if the initial values and bounds are identical. In future, we could upgrade to option 2, but 1 seems the more controlled approach for now.